### ISSUE TITLE
[proposal] Allow query of a key fragment as a document

### ISSUE THREAD

**Issue by F-Node-Karlsruhe** [decentralized-identity/did-resolver#141]
Similar to issue https://github.com/decentralized-identity/web-did-resolver/issues/121
**Is your feature request related to a problem? Please describe.**
I'm currently using the web-did-resolver in a custom document loader and it works very well, but i still have to query key fragments separately like so:

```ts
        const [did, keyFragment] = url.split('#')

        // fetch document
        const didDocument: any = (await getResolver().resolve(url)).didDocument

        // if a keyFragment of the DID document is queried
        if (keyFragment && didDocument) {
            
            // TODO allow query of other fields than verification method as well
            const keyFragmentDoc: any | undefined = didDocument.verificationMethod.filter(function(method: any) {
                                                                            return method.id === url || method.id === verificationMethod;
                                                                        })[0];

            if (!keyFragmentDoc)
                throw new jsonldSignatures.VerificationError(
                            new Error(`${verificationMethod} is an unknown key fragment for ${did}`)
                            );

            return {
                contextUrl: null,
                documentUrl: url,
                // deliver keyFragment with the DID doc context
                document: Object.assign(verificationMethodDoc, {'@context': didDocument['@context']}),
            };

        }

        return {
            contextUrl: null,
            documentUrl: url,
            document: didDocument,
        };
```

**Describe the solution you'd like**
If the key fragment is specified the resolver should not return the DID document, but the key fragment document. (See code above)

**Describe alternatives you've considered**
See code above

**Comment by peacekeeper** [comment]
I agree this would be a useful feature. This is part of a bigger concept called DID URL Dereferencing, which is partially explained in the DID Core spec (see [here](https://www.w3.org/TR/did-core/#did-url-dereferencing)), and partially in the DID Resolution spec (see [here](https://w3c-ccg.github.io/did-resolution/#dereferencing-algorithm-secondary)). There are also some test cases in the DID Resolution test suite (see [here](https://github.com/w3c-ccg/did-resolution-test-suite)).

There are some things to consider, e.g. the representation of the DID document (JSON-LD or JSON or CBOR etc.) and relative references. In general URI architecture, dereferencing a fragment depends not on the URI scheme (i.e. "did:"), but on the content type of the deferenced primary resource (i.e. the DID document).

Anyway, this is just some background information, and not all the details are "officially" specified yet. In general, +1 to supporting the dereferencing of fragments in DID resolution libraries like this one.
